### ISSUE TITLE
Realtime Updates

### ISSUE THREAD

**Issue by PatchworkBoy** [PatchworkBoy/homebridge-edomoticz#29]
Need to add ability to subscribe to Domoticz’ native MQTT channel (domoticz/out flat architecture), parse and then run setValue against matched accessory after checking dtype in the stream to decide whether to use nvalue or svalue etc

This will allow (pretty much) realtime updates rather than manually having to initialise an update of sensor data.

Intend to use (and package.json is now modified to pull in) https://www.npmjs.com/package/mqtt ^1.0.0 unless a more suited alternative is found.

This side of things will take a while to evolve as we’ll need to add relevant bits into config.json to cover separate mqtt authentication etc.

All contributions should be pulled against https://github.com/PatchworkBoy/homebridge-eDomoticz/tree/mqtt-integration

**Comment by PatchworkBoy** [comment]
Current state of the matt-integration branch is:

Connects to mqtt://<server>:<port>, subscribes to domoticz/out stream, echoes messages out to the Homebridge log.

Messages need parsing and acting upon.

**Comment by PatchworkBoy** [comment]
Things to watch out for...

On Raspberry Pi B+ the mqtt logging currently causes memory usage to increase (or it does for me, but I have a lot of accessories spitting updates out every second or so). This leaves the Pi B+ on average with 8Mb free. Once this gets used up, Raspbian kills homebridge (and forever then restarts it). Can usually see this as a SIGPIPE error in the forever log.

**Comment by PatchworkBoy** [comment]
Consider this the ReleaseCandidate version, currently in testing towards Release status: https://github.com/PatchworkBoy/homebridge-eDomoticz/tree/mqtt-integration

To use, firstly it’s presumed that users have Mosquito or another MQTT broker running locally on the same system as Domoticz. It’s preinstalled on both Domoticz SD Card images.

Now, In Domoticz web interface: Setup > Hardware...

Add New Hardware...

Type: MQTT Client Gateway with LAN interface
Remote Address: 127.0.0.1
Port: 1883
Username: (only populate these if you’ve set up security in your MQTT broker)
Password: (only populate these if you’ve set up security in your MQTT broker)
Publish Topic: out

Domoticz will now publish all device changes via MQTT.

Amend your homebridge config.json to include the MQTT fields as follows:

```
{
    "bridge": {
        "name": “Homebridge”,
        "username": "CC:21:3E:E4:DE:33”,
        "port": 51826,
        "pin": "031-45-154”
    },
    "description": "Configuration file for (e)xtended Domoticz platform.”,
    "platforms": [
        {
            "platform": “eDomoticz”,
            "name": “eDomoticz”,
            "server": “127.0.0.1”,
            "port": “8080”,
            "ssl": 0,
            "roomid": 0,
            "mqttenable": 1,
            "mqttserver": “127.0.0.1”,
            "mqttport": “1883”,
            "mqttauth": 1,          //only needed if you've password protected mosquitto via mosquitto.conf
            "mqttuser": "username", //only needed if you've password protected mosquitto via mosquitto.conf
            "mqttpass": "password"  //only needed if you've password protected mosquitto via mosquitto.conf
        }
    ],
    "accessories": []
}
```

SWITCHES are now realtime updated on the fly via HomeKit... ie: Devices of type Switch, Contact, Blinds, Smoke, BlindsInverted, Dimmer, Motion, PushOn, BlindsPercentage & BlindsPercentageInverted.

SENSORS are not. These are a work-in-progress. Priority was on the switches.

**Comment by xTheRamon** [comment]
Closing this, just merged `mqtt-integration` into `master` (466db94)
